subsys/net/ip/net_context.c

```diff
@@ -619,6 +708,12 @@ int net_context_listen(struct net_context *context, int backlog)
 	}
 #endif /* CONFIG_NET_OFFLOAD */
 
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_RAW) &&
+	    net_context_get_family(context) == AF_PACKET) {
+		net_context_set_state(context, NET_CONTEXT_LISTENING);
```

Comment:
Can RAW sockets be listening at all? UDP sockets can be listening, and RAW is even more hardcore than that.